### PR TITLE
Rewards Quick actions 

### PR DESCRIPTION
**Pull request by pixelfact**

### CODE REVIEW

File: src/components/TagButton/TagButton.tsx [hunk 1]
```diff
@@ -4,7 +4,6 @@ import { MouseEvent } from "react";
 
 export default function TagButton({ children, onClick, ...rest }: ButtonHTMLAttributes<HTMLButtonElement>) {
   const handleClick = (e: MouseEvent<HTMLButtonElement>) => {
-    e.preventDefault();
```
Comment (haydencleary) [inline src/components/TagButton/TagButton.tsx:7]: Why was this removed ? It might be a good idea to leave it to avoid this button submitting any form it may be inside of.

File: src/pages/Projects/SubmitProject/index.tsx [hunk 2]
```diff
@@ -38,6 +38,7 @@ export default function SubmitProject() {
   };
 
   const onClick = () => {
+    console.log("onClick", modalOpened);
```
Comment (haydencleary) [inline src/pages/Projects/SubmitProject/index.tsx:41]: Don't forget to remove this ;)

File: src/types.ts [hunk 3]
```diff
@@ -158,6 +158,12 @@ export enum GithubContributionType {
   CodeReview = "CODE_REVIEW",
 }
 
+export enum GithubContributionShortenTypeLabel {
+  Issue = "Issue",
+  PullRequest = "PR",
+  CodeReview = "Code Review",
+}
+
```
Comment (haydencleary) [inline src/types.ts:166]: Why is this necessary ? Won't the current type do ?

File: src/pages/ProjectDetails/Rewards/RewardForm/AutoAdd/QuickActionsMenu.tsx
Comment (haydencleary) [inline src/pages/ProjectDetails/Rewards/RewardForm/AutoAdd/QuickActionsMenu.tsx:1]: Perhaps you could make a generic Menu component that handles the styles and main behaviour then refactor this for your use case. I can imagine other menus coming up in the application.

File: src/pages/ProjectDetails/Rewards/RewardForm/AutoAdd/QuickActionsMenu.tsx [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+        enterFrom="transform scale-95 opacity-0"
+        enterTo="transform scale-100 opacity-100"
+        leave="transition duration-75 ease-out"
+        leaveFrom="transform scale-100 opacity-100"
+        leaveTo="transform scale-95 opacity-0"
+        className="relative z-10"
+      >
+        <Menu.Items className="absolute -left-6 -top-12 z-40 w-auto min-w-max origin-top-right rounded-xl bg-greyscale-900 p-2 font-walsheim text-sm font-normal shadow-medium outline-none focus:outline-none">
+          <div className="cursor-pointer rounded p-2 hover:bg-white/5">
+            <Menu.Item as="div" onClick={handleAdd}>
```
Comment (haydencleary) [inline src/pages/ProjectDetails/Rewards/RewardForm/AutoAdd/QuickActionsMenu.tsx:40]: Menu items should probably be buttons for accessibility and semantic reasons.

Comment (pixelfact) [reply src/pages/ProjectDetails/Rewards/RewardForm/AutoAdd/QuickActionsMenu.tsx:40]: i had a propagation side effet when using a button even with stopPropagation and preventDefault !!

File: src/components/TagButton/TagButton.tsx (hunk 1 above)
Comment (pixelfact) [reply src/components/TagButton/TagButton.tsx:7]: when I pass the button as children of ActionMenu the onclick function is blocked because of the preventDefault

**Review by haydencleary — APPROVED** [review]
